File: backend/app/schemas/health.py
"""Health check schema."""
from pydantic import BaseModel
from typing import Optional


class HealthResponse(BaseModel):
    """Health check response with all system resource information."""
    status: str
    timestamp: Optional[str] = None
    
    # CPU Information
    cpu_percent: Optional[float] = None
    cpu_count: Optional[int] = None
    cpu_frequency_mhz: Optional[float] = None
    cpu_frequency_max_mhz: Optional[float] = None
    
    # Memory Information
    memory_total_gb: Optional[float] = None
    memory_available_gb: Optional[float] = None
    memory_used_gb: Optional[float] = None
    memory_percent: Optional[float] = None
    
    # Swap Information
    swap_total_gb: Optional[float] = None
    swap_used_gb: Optional[float] = None
    swap_percent: Optional[float] = None
    
    # Disk Information
    disk_total_gb: Optional[float] = None
    disk_used_gb: Optional[float] = None
    disk_free_gb: Optional[float] = None
    disk_percent: Optional[float] = None
    
    # Network Information
    network_bytes_sent: Optional[int] = None
    network_bytes_recv: Optional[int] = None
    network_packets_sent: Optional[int] = None
    network_packets_recv: Optional[int] = None
    
    # Process Information
    process_memory_mb: Optional[float] = None
    process_cpu_percent: Optional[float] = None
    process_num_threads: Optional[int] = None

File: frontend/src/types/health.ts
/**
 * Health check response types
 */

export interface ComponentHealth {
  status: "healthy" | "unhealthy" | "unavailable" | "degraded";
  message?: string;
  error?: string;
}

export interface HealthResponse {
  status: "healthy" | "degraded" | "unhealthy";
  timestamp?: string;
  
  // CPU Information
  cpu_percent?: number;
  cpu_count?: number;
  cpu_frequency_mhz?: number;
  cpu_frequency_max_mhz?: number;
  
  // Memory Information
  memory_total_gb?: number;
  memory_available_gb?: number;
  memory_used_gb?: number;
  memory_percent?: number;
  
  // Swap Information
  swap_total_gb?: number;
  swap_used_gb?: number;
  swap_percent?: number;
  
  // Disk Information
  disk_total_gb?: number;
  disk_used_gb?: number;
  disk_free_gb?: number;
  disk_percent?: number;
  
  // Network Information
  network_bytes_sent?: number;
  network_bytes_recv?: number;
  network_packets_sent?: number;
  network_packets_recv?: number;
  
  // Process Information
  process_memory_mb?: number;
  process_cpu_percent?: number;
  process_num_threads?: number;
  
  // Component Health Status
  database_status?: string;
  redis_status?: string;
  embedding_service_status?: string;
  components?: Record<string, ComponentHealth>;
}

export interface PerformanceMetrics {
  total_requests: number;
  total_errors: number;
  average_response_time_seconds: number;
  status_codes: Record<string, number>;
  top_endpoints: Array<{
    endpoint: string;
    count: number;
    avg_time_seconds: number;
    total_time_seconds: number;
  }>;
}

